From d0f302751a57cea1ee64261c749f59c4026f7af7 Mon Sep 17 00:00:00 2001
From: Changsheng Wu <wuchangsheng2@huawei.com>
Date: Thu, 16 Dec 2021 11:46:58 +0800
Subject: [PATCH] 
 huawei-0008-dpdk-fix-cpu-flag-error-in-Intel-R-Xeon-R-CPU-E5-262

---
 config/meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/config/meson.build b/config/meson.build
index 17b5bec406..293703a90d 100644
--- a/config/meson.build
+++ b/config/meson.build
@@ -121,6 +121,10 @@ if cpu_instruction_set == 'generic'
     elif host_machine.cpu_family().startswith('ppc')
         cpu_instruction_set = 'power8'
     endif
+elif host_machine.cpu_family().startswith('x86')
+    if cc.get_define('__SSE4_2__', args:'-march=native') == ''
+	cpu_instruction_set = 'corei7'
+    endif
 endif
 
 dpdk_conf.set('RTE_MACHINE', cpu_instruction_set)
-- 
2.27.0

